x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/SecurityIndexRolesMetadataMigrationIT.java

```diff
@@ -6,54 +6,80 @@
  */
 package org.elasticsearch.upgrades;
 
+import org.elasticsearch.client.Node;
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.RequestOptions;
 import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.client.RestClient;
 import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.core.Nullable;
 import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
 
 import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+import static org.elasticsearch.xpack.core.security.action.UpdateIndexMigrationVersionAction.MIGRATION_VERSION_CUSTOM_DATA_KEY;
+import static org.elasticsearch.xpack.core.security.action.UpdateIndexMigrationVersionAction.MIGRATION_VERSION_CUSTOM_KEY;
+import static org.elasticsearch.xpack.core.security.test.TestRestrictedIndices.INTERNAL_SECURITY_MAIN_INDEX_7;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
 
 public class SecurityIndexRolesMetadataMigrationIT extends AbstractUpgradeTestCase {
 
-    public void testMetadataMigratedAfterUpgrade() throws Exception {
-        String testRole = "test-role";
-        String metaKey = "test_key";
-        String metaValue = "test_value";
-
-        Map<String, Object> testMetadata = Map.of(metaKey, metaValue);
+    public void testRoleMigration() throws Exception {
+        String oldTestRole = "old-test-role";
```

Comment:
I think keeping this test focused and using the same meta key and value might make it easier to parse and leave the testing of the query logic to the `QueryRoleIT` suite.

Reply:
Pushed [d225966](https://github.com/elastic/elasticsearch/pull/108733/commits/d225966f37c771880f5aa5b446e21aa1741762d6)